src/libraries/System.Web.HttpUtility/src/System/Web/Util/HttpEncoder.cs

```diff
@@ -38,7 +38,7 @@ private static void AppendCharAsUnicodeJavaScript(StringBuilder builder, char c)
             }
 
             // Don't create string writer if we don't have nothing to encode
-            int pos = IndexOfHtmlAttributeEncodingChars(value, 0);
+            int pos = IndexOfHtmlAttributeEncodingChars(value);
             if (pos == -1)
```

Comment:
```suggestion
            if (pos < 0)
```
while you are on it?
(other places as well)

Reply:
why?
it is accurate because api docs say IndexOfAny returns -1 if no character was found.
x86 mac code for `== -1` is generated as `cmp/sete` and for `< 0`, it's `cmp/setl`. both have identical uops and tp.

Reply:
> docs say IndexOfAny returns -1 if no character was found

Yep. But as a found index can only be in the range $[0, length)$ a comparions with `0` is also correct.

> x86 mac

I don't know about the codegen for mac, but in general for x86/x64 comparisons with `0` emit `test/jmp`-combo which can be fused at cpu-level into one instruction that gets actually executed. For .NET 7 (?) some other branch-less optimizaitons are enabled, as [this sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHgIIAGwYBZcgAoAlgDsMDSQBMEASgYBeAHzylDAIRqGccgG4uPeg35DRpKbO0r1WxQgYaDABlPUevSwOEROjs5F1VNBwYAHgYvGgBfIA=) shows (sharplab was fast with the update to .NET 7 this time, hooray).

For the concrete use-case here the difference won't show up for sure -- especially as bit down the code an object is allocated -- but the change is simple, and every tiny little bit (that's easy to change) counts 😉.

Reply:
> I don't know about the codegen for mac,

i meant the machine code not macintosh :)

> For .NET 7 (?) some other branch-less optimizaitons are enabled

nice! `< 0` is better for int comparison. i was using a wrong [sharplab sample](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8BiAOwFcAbS7YSmAAgBMBLXW+gWACgABAJgYBhHgG8eDSQ14BmBsAgRKDALIBGABS81ABga4AlAwC8APn0A6ADIxSAcwwALE8YZw1AbglTZ8xcpV+LV19IzNLG3snBgAeBh0PKR4AXyA===) with string.Length and looking up their perf characteristics in https://uops.info/table.html (sete is not available in the table but setz and setl are there; sete and setz are the same)

> sharplab was fast with the update to .NET 7 this time, hooray

yay! godbolt is also on its way https://github.com/compiler-explorer/infra/pull/869